src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheckTest.java

```diff
@@ -529,6 +529,21 @@ public void testUnusedMethod() throws Exception {
             .isEqualTo("CATCH_FRAME");
     }
 
+    @Test
+    public void testUnusedMethodFor() throws Exception {
```

Comment:
New surviving from cascades.
https://github.com/checkstyle/checkstyle/actions/runs/3903501555/jobs/6667951329#step:6:418

I assume it cascade from https://github.com/checkstyle/checkstyle/pull/12628#discussion_r1068222005 .

Reply:
as we agreed in private chat....
lets keep survival not a hacky junit.

sadly , there is no ability to make comment in generated xml files to warn users to not play with this survival, as it too expensive for now.

what we can do .... is keep such test methods with comment "until XXXXX" to remove them and convert to suppression if suppression file is about to be removed.

Reply:
> sadly , there is no ability to make comment in generated xml files to warn users to not play with this survival, as it too expensive for now.

Why can't we make comment in xml file?

Reply:
we do `diff` on generated and git file. 
we can potentially remove all comments from files before diff. But it will make extra complication for people on local, `git diff` will not work anymore.

Reply:
> we do diff on generated and git file

Where do we do this? One of the reasons for migrating to xml file was to be able to place comments and have permanent suppression list.

Reply:
We have comments in files already: https://github.com/checkstyle/checkstyle/blob/72fa0536c643e56934cb7095061150ebf17a84e6/config/pitest-suppressions/pitest-common-2-suppressions.xml#L183

Reply:
@romani I think we are going to have some other issues because of git diff. We don't validate git diff on a success. Someone could put things in the wrong order and it will cause a weird git diff when some failure does occur and the user copies the git diff.

Reply:
We try to validate that after execution of CI , git state is not changed.

If we missed it somewhere we should fill gap.

We can make single issue to re-review pure unit tests after we finish with pitest.

Reply:
Comment on the method? I thought we were saying to remove this method and leave suppression?

Reply:
I just try to think how to avoid anyone else to try to tackle it.

Reply:
We do not run `git diff` on pitest: https://github.com/checkstyle/checkstyle/actions/runs/3924059900/jobs/6708062064

>We try to validate that after execution of CI , git state is not changed.

> If we missed it somewhere we should fill gap.

We do not write to file automatically, git diff validation will have no affect on this. We need to be able to place comments on suppressions, this is one of the advantages of new suppression system.

>We can make single issue to re-review pure unit tests after we finish with pitest.

We agreed that suppression is better than pure UTs for difficult mutations, we have ability to leave comments on suppressions, we should do it.

Please create a permanent suppression list at the bottom of the suppression file and move this one there.

Reply:
update of files is in progress - https://github.com/checkstyle/checkstyle/pull/12614

Reply:
@romani @nrmancuso Please communicate with each other and let me know which way you both want to handle this as you both gave conflicting changes requested.

Reply:
Whatever @romani wants to do is fine, let's get this merged.

Reply:
@rnveach , please make a comment above to explain reason. And may be create ticket that we will use to as single marker to quickly find such compromise unit tests.

Reply:
Done.